### PR TITLE
VZ:8147: ManualScalerTrait Issue

### PR DESCRIPTION
**Pull request by neeraj-n-singh**

### CODE REVIEW

**Review by mbenmclean — APPROVED** [review]

File: pkg/oam/util/helper.go [hunk 1]
```diff
@@ -261,7 +261,8 @@ func fetchChildResources(ctx context.Context, mLog logr.Logger, r client.Reader,
 // PatchCondition condition for a conditioned object
 func PatchCondition(ctx context.Context, r client.StatusClient, workload ConditionedObject,
 	condition ...cpv1alpha1.Condition) error {
-	workloadPatch := client.MergeFrom(workload)
+	patch := workload.DeepCopyObject().(client.Object)
```
Comment (kminder) [inline pkg/oam/util/helper.go:266]: This needs a comment about why it is required.

File: pkg/oam/mock/mocks.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 
-	"github.com/crossplane/crossplane-runtime/apis/common/v1"
+	v1alpha1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
```
Comment (kminder) [inline pkg/oam/mock/mocks.go:33]: This isn't alpha1, right?

Comment (neeraj-n-singh) [reply pkg/oam/mock/mocks.go:33]: Yes, it's an alias to reduce the code changes from the original fork.

File: pkg/oam/util/helper.go (hunk 1 above)
Comment (neeraj-n-singh) [reply pkg/oam/util/helper.go:266]: Added.

**Pull request merged**